include/reshade_events.hpp

```diff
@@ -1564,6 +1564,15 @@ namespace reshade
 		/// </remarks>
 		reshade_reorder_techniques,
 
+		/// <summary>
+		/// Called when the ReShade overlay is about to be activated (shown) or deactivated (hidden).
+		/// <para>Callback function signature: <c>bool (api::effect_runtime *runtime, bool* activate, api::input_source source)</c></para>
+		/// </summary>
+		/// <remarks>
+		/// To prevent overlay activation, set the value of activate and return <see langword="true"/>, otherwise return <see langword="false"/>.
+		/// </remarks>
+		reshade_overlay_activation,
```

Comment:
I'd rename this to `reshade_open_overlay`, so to be more consistent with a uniform source that exists doing something similar called `overlay_open` (which can be used to check whether the overlay is open in effects, https://github.com/crosire/reshade/blob/main/source/runtime.cpp#L1664).
And also make it behave more similar to other action events like `addon_event::draw`, in that returning true will make ReShade skip the open/close request, rather than requiring to modify a passed in parameter:
```c++
bool (api::effect_runtime *runtime, bool open, api::input_source source)
```